### ISSUE TITLE
Terribly frustrating :( All encrypted chats showing unable to decrypt after sign-in

### ISSUE THREAD

**Issue by jittygitty** [vector-im/element-meta#2029]
### Steps to reproduce

1. Where are you starting? What can you see?
I signed-in after being "forced signed-out" when app complained browser deleted files due to low on space, and when I signed back-in and gave security key, all the "encrypted chats" are full of "unable to decrypt message", unencrypted chats work.

#### What did you expect?
I would expect that the chat files would NOT be "deleted" and be forced signed out.
#### What happened instead?
But if that can't be avoided, I would expect app to work properly after I signed back in and verified with security key etc, yet encrypted chats say "unable to decrypt..."

UPDATE:
I did a right click on element and quit it but when I try to re-open it now, it doesn't work, won't open or maybe crashes trying to open, nothing comes up etc. Another install that uses a 'portable' element desktop and connect to different account and different server (connects to main matrix.org server) does work fine still and was not affected at all by the low space issue it seems. So currently I'm stuck and can't open/start my main element install, i may need to try and  re-install/upgrade etc.

### Operating system

Windows 10

### Application version

Element version: 1.11.38 Olm version: 3.2.14

### Homeserver

Dendrite server, maybe give a good easy way to see version?

### Will you send logs?

No

**Comment by jittygitty** [comment]
Update: I was able to get the main 'installed' element app to "open/run" again only by re-installing again via a fresh downloaded version from official element chat website. But all encrypted chats still say "unable to decrypt message"...again, my other "portable" element chat that ran side by side with this other one, that portable one was NOT affected by the windows 10 low space which apparently deleted chat files and forced "sign-out" of my main installed element chat app causing all these issues.

**Comment by jittygitty** [comment]
Update: Seems like I do have some encrypted channels/rooms that have messages visible, but next to them they have an icon that on hover says either "The authenticity of this encrypted message can't be guaranteed on this device" or less often "encrypted by a deleted session" ... Also in some "encrypted channels", I noticed a few messages with red icon that on hover say "Unencrypted" (but this might have been there before this forced "sign-out" problem though) ....most of my other messages in there say unable to decrypt message, and the other user's say that authenticity can't be guaranteed, but I can see them.

**Comment by jittygitty** [comment]
UPDATE: I got on computer today (PC was either awake and app running or maybe standby/sleeping not sure) and those encrypted rooms that previously showed "unable to decrypt message", now all of the sudden the messages are visible, but each message has that grey shield with the message "The authenticity of this encrypted message can't be guaranteed on this device".

So, on the bright side, the app seems like it tried to fix itself through some resyncing of keys or something, not sure how else to explain the messages visible now...
I am guessing those "authenticity" complaints are benign and can be ignored and the messages are still all encrypted etc, correct?

**Comment by jittygitty** [comment]
Update: Now in one of the encrypted rooms, direct message chat with one person, that my previous messages they can see fine but new messages I send simply say "Waiting for this message, this may take a while", but I can talk with them fine in another room.

In fact in the other room, that grey shield with the "authenticity" complaint seems to have magically gone away on its own and they can see my messages there. But in this direct message room with only us two members, when i send a message they just get "Waiting for this message, ...

They are using Android Client, I am using Element Desktop. Ctrl-Shit-I debug shows:

Failed to load resource: the server responded with a status of 404 (Not Found)
rageshake.ts:74 Failed to query SIP identity for user M_NOT_FOUND: MatrixError: [404] The Matrix User ID was not found (https://my.dendritematrixserver.com/_matrix/client/r0/thirdparty/user/im.vector.protocol.sip_virtual?native_mxid=%40justafriend%3Amy.dendritematrixserver.com)
    at c (vector://vector/webapp/bundles/e1d4bea…/bundle.js:2:1355935)
    at p.requestOtherUrl (vector://vector/webapp/bundles/e1d4bea…/bundle.js:2:3601620)
2
rageshake.ts:74 Stopping queue 'message' as it is now empty
my.dendritematrixserver.com/_m…my.dendritematrixserver.com:1 
 Failed to load resource: the server responded with a status of 404 (Not Found)
rageshake.ts:74 Failed to query SIP identity for user M_NOT_FOUND: MatrixError: [404] The Matrix User ID was not found (https://my.dendritematrixserver.com/_matrix/client/r0/thirdparty/user/im.vector.protocol.sip_virtual?native_mxid=%40justafriend%3Amy.dendritematrixserver.com)
    at c (vector://vector/webapp/bundles/e1d4bea…/bundle.js:2:1355935)
    at p.requestOtherUrl (vector://vector/webapp/bundles/e1d4bea…/bundle.js:2:3601620)

**Comment by kazin-kharizma** [comment]
Hey @jittygitty did anybody get back to you about this issue? I have now reported it as well but I notified you triaged it 4 days ago. It is more widespread now it seems and here is my report: https://github.com/vector-im/element-web/issues/26011#issue-1857967153

@dbkr perhaps you could merge the two issue together as I’ve begun to reference mine across a number of channels and chats and people are looking for updates. It would give some pieces of mind knowing that it’s being examined. I also have my log reports I can send but I’d prefer not to post to GitHub directly.

**Comment by jittygitty** [comment]
Hi @kazin-kharizma  I have no outside communication other than here on this, so no I have not heard anything from anyone. In my case I do not think it has anything to do with https://status.matrix.org/incidents/0xrbdq5kp1s9

Because in my case, when this happened it happened when my C drive was very low on space, and my "portable" version of Element Chat which had my account on "matrix.org" server, was not affected at all and worked fine. Only my regularly installed Element Chat was affected, and I think this is the main reason, because it was controlled by "Windows" as an installed app and windows had some power over its cache mechanisms. In fact, the app even notified me at some point that the Browser had deleted some things or something like that due to lack of space.

Since the other install was "portable" and wasn't under control of Windows management etc, it was not affected. But when issue started, i first noticed that the app was complaining it could not talk to my dendrite server. But I logged in to server and saw that my dendrite server seemed to be working just fine. The app then ended up signing me out etc and those problems appeared as I described above.

IMPORTANT POINT: The biggest issue isn't that the installation was messed up due to Browser/Windows deleting files, forcing sign-ff and then forcing re-installation of the app. The big problem is that even after all that, re-installing, re-importing my key etc, that the app still didn't "recover" correctly and was still having communication errors. 

(On a separate note, I think I complained before (but nobody agreed with my concern?) that the APP should NEVER sign me out, UNLESS I tell it its' ok to do so.

I had an issue before where during dendrite server upgrade, I made some mistake and restarted the server with wrong sql files or something, can't remember exactly, and then my Element Chat app "FORCED" signed me out, without my permission.)

Even now I still have some chats where on my device on element chat, my messages have the grey shield saying: "The authenticity of this encrypted message can't be guaranteed on this device". But ironically the other person's messages in this direct chat don't have this message. So my own device says it has more trouble with my own messages typed on my device...I would've expected more issues authenticating the other person's messages not my own : D 

In some other rooms, all the messages have that grey shield and authenticity complaint, but as I said in previous message, i had issues sending messages to some channels...

To be honest, I am now looking to switch to another chat until Element Chat can mature in its encryption handling and hopefully when it can offer either live-sync of server or the p2p version it had promised long ago where every client could run its own "local server" so that all of our chat logs can always be with us locally on our device/client where we can access EVERYTHING even when INTERNET is not available.

Currently if you lose internet you can't access anything from your chats, that's worse than email where we have IMAP/pop etc. And if server dies, you lost everything unless you have backups. And it's not simple doing live backups of in use database.

Getsession was interesting but too slow for everyday use and they need to make the entire app "portable" so you can grab the folder and move it to another device.

I think simplex-chat is interesting, all your data local to your device, and server is just a "dumb" relay it seems. But their LICENSE of AGPL is a stopper for me, I like to build my own custom versions for friends/family and I won't be bothered with AGPL, see: https://github.com/simplex-chat/simplex-chat/issues/2869

Anyway I really like Element Chat, so I look forward to:
1. Always available FULL access to ALL my CHAT data/logs, "locally" on my client.
2. Multi-Account, multiple profiles with one installation.
3. Chat encryption to work better and not break things so often.

Sadly there isn't something that I've found yet that foots the bill so to speak...

**Comment by jittygitty** [comment]
UPDATE: yet more demoralizing news...

I ask an IPHONE user if he saw my recent messages in a channel with a handful users all connected to my server. He texts me back saying that:

"your chat sucks" it "never works"

Then he sends a screenshot which shows "**Unable to decrypt: The sender's session has not sent us the keys for this message. ** Re-request encryption keys from your other sessions."  That's what's shown for my messages to him in that channel.

In the screenshot he sent, his attempted sent messages just show the error: **Unable to decrypt: The operation couldn't be completed. (MatrixSDKCrypto.DecryptionError error 2.)**" messages failed to send...retry button... (Why is he getting a DECRYPTION error when trying to "send" a message? Is his client unable to decrypt his own msg??)

The encryption I think really needs to be "simplified" if it's so difficult to get it working. 

Can someone please enlighten me why we can't simply offer some sort of "challenge/response" method of sending/sharing decryption keys to fix such errors?

I'd love to continue using element if I hear this is going to be all resolved soon, but if this is going to be something we just have to live with, as much as I hate to start again, if basic text chat can't be fixed, I guess will have to try rocketchat/zulip/mattermost etc.

When element actually does work, it's great for the most part, so I hope there's "hope".

**Comment by jittygitty** [comment]
UPDATE: I now started up that "portable element" version of element chat, and was greeted with:
"Missing session data
Some session data, including encrypted message keys, is missing. Sign out and sign in to fix this, restoring keys from backup.
Your browser likely removed this data when running low on disk space. To help us prevent this in future, please send us logs." 

And the only option was a green "sign out" button. 
I just killed the app, since I didn't want to click the sign out button.

But after I reopened the portable element chat, I am greeted with a sign-in page, apparently it signed me out anyway, even though I never clicked to sign-out. This portable element chat was always connected to main matrix.org public server.

There's plenty of space now but I guess previously when ran low even this portable version got its keys caches etc deleted it seems. Is there no way to prevent it?

Anyway, I'm still a lot more concerned though with the encryption/decryption errors that the Iphone user was having, which I mentioned in my previous post. It makes people give up on using this and go back to their whatsapp or facetime or whatever.

**Comment by richvdh** [comment]
There are an awful lot of words here, and I'm afraid it's rather too hard to follow what is going on.

I appreciate that it's frustrating when the encryption gets in your way. We're tracking the causes of that at #245.

I'm afraid I'm going to close this issue as it doesn't seem to be very actionable right now. Feel free to open more specific issues clearly describing individual problems. Please be aware that screenshots can be very helpful when explaining what you are seeing; also be aware that normally we will need you to submit debug logs from your client in order to be able to investigate any failures.

**Comment by jittygitty** [comment]
@richvdh Yea apologies for the long thread, but the quite "simple" take-away is please come up with a way so that Windows does not treat Element Chat as a simple browser and give itself authority to "delete" related metadata or caches that logs/signs you out.
